internal/debug/doc.go

```diff
@@ -0,0 +1,18 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*
+Package debug provides access to enable internal debugging of opentelemetry.
```

Comment:
should this package be internal? don't we want to use it e.g. in go-contrib?

Reply:
Not yet.  That would significantly expand the scope of this PR.  The idea is something that can be used internally by the SDK.  If we make this an exposed API for contrib, then we also expose it to end-users, which brings up a whole different set of concerns.  Things like:

- Should an instrument wrapper use the SDK internal error handling or the library's error handling?
- Would library writers adopt this logging system when we will want to push to use the "(un)official" logging API that goes through the SDK's processing steps?
- Is this API robust enough to cover most users' use cases? (my guess is no)

In the end, the only misses that are glaring are any contrib exporters.